File: tools/python-incident-analyzer/src/incident_analyzer/domain/incident_severity.py
"""Incident severity level."""

from __future__ import annotations

from enum import StrEnum


class IncidentSeverity(StrEnum):
    """Incident severity mapped from the Java backend."""

    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"

    @classmethod
    def from_json(cls, value: str) -> IncidentSeverity:
        """Parse a backend severity value into an enum."""
        try:
            return cls(value)
        except ValueError as exc:
            raise ValueError(f"Unknown incident severity: {value}") from exc

File: frontend/flutter-rules-practice/lib/features/incidents/domain/incident_severity.dart
enum IncidentSeverity {
  low,
  medium,
  high,
  critical;

  static IncidentSeverity fromJson(String value) {
    return switch (value) {
      'LOW' => IncidentSeverity.low,
      'MEDIUM' => IncidentSeverity.medium,
      'HIGH' => IncidentSeverity.high,
      'CRITICAL' => IncidentSeverity.critical,
      _ => throw ArgumentError('Unknown incident severity: $value'),
    };
  }

  String toJson() {
    return switch (this) {
      IncidentSeverity.low => 'LOW',
      IncidentSeverity.medium => 'MEDIUM',
      IncidentSeverity.high => 'HIGH',
      IncidentSeverity.critical => 'CRITICAL',
    };
  }

  String get label {
    return switch (this) {
      IncidentSeverity.low => 'Low',
      IncidentSeverity.medium => 'Medium',
      IncidentSeverity.high => 'High',
      IncidentSeverity.critical => 'Critical',
    };
  }
}